tests/Unit/ActionTesting.hpp

```diff
@@ -61,6 +61,260 @@ struct get_array_index;
+ * actions that are called during the `TestRepeatUntil` phase. The
+ * `Initialization` phase action list now has the action
+ * `ActionTesting::InitializeDataBox`, which takes simple tags and compute tags
+ * that will be added to the DataBox in the initial phase. How the values for
+ * the simple tags are set will be made clear below when we discuss the
+ * `ActionTesting::emplace_component_and_initialize()` function. The action
+ * lists for the `TestGoto` and `TestRepeatUntil` phases are fairly simple and
+ * not the focus of this discussion.
+ *
+ * Having discussed the metavariables and component, let us now look at the test
+ * case.
+ *
+ * \snippet Test_Goto.cpp test case
+ *
+ * Just as for the `TerminatePhase` test we have some type aliases to reduce the
+ * amount of typing needed and to make the test easier to read. The runner is
+ * again created with the default constructor. However, the component is now
+ * inserted using the `ActionTesting::emplace_component_and_initialize()`
+ * function. The third argument to `emplace_component_and_initialize` is a
+ * `tuples::TaggedTuple` of the simple tags, which is to be populated with the
+ * initial values for the component. Note that there is no call to
+ * `ActionTesting::set_phase(&runner, Metavariables::Phase::Initialization)`:
+ * this and the required action invocation is handled internally by
+ * `ActionTesting::emplace_component_and_initialize()`.
+ *
+ * Once the phase is set the next action to be executed is set to be
+ * `Actions::Label<Label1>` by calling the
+ * `MockRuntimeSystem::force_next_action_to_be()`  member function of the
+ * runner. The argument to the function is the array index of the component for
+ * which to set the next action. After the `Label<Label1>` action is invoked we
+ * check that the next action is the fourth (index `3` because we use zero-based
+ * indexing) by calling the `MockRuntimeSystem::get_next_action_index()` member
+ * function. For clarity, the indices of the actions in the `Phase::TestGoto`
+ * phase are:
+ * - 0: `Actions::Goto<Label1>`
+ * - 1: `Actions::Label<Label2>`
+ * - 2: `Actions::Label<Label1>`
+ * - 3: `Actions::Goto<Label2>`
+ *
+ * ### DataBox introspection
+ *
+ * Since the exact DataBox type of any component at some point in the action
+ * list may not be known, the `ActionTesting::get_databox_tag()` function is
+ * provided. An example usage of this function can be seen in the last line of
+ * the test snippet above. The component and tag are passed as template
+ * parameters while the runner and array index are passed as arguments. It is
+ * also possible to retrieve DataBox directly using the
+ * `ActionTesting::get_databox()` function if the DataBox type is known:
+ *
+ * \snippet Test_ActionTesting.cpp get databox
+ *
+ * It is also possible to check if an item can be retrieved from the DataBox
+ * using the `ActionTesting::tag_is_retrievable()` function as follows:
```

Comment:
maybe mention the mutable version that allows editing the databox if you know the type.